src/gp_activetable.c

```diff
@@ -1182,42 +1177,23 @@ pull_active_table_size_from_seg(HTAB *local_table_stats_map, char *active_oid_ar
 
 		for (j = 0; j < PQntuples(pgresult); j++)
 		{
-			reloid     = atooid(PQgetvalue(pgresult, j, 0));
-			tableSize  = (Size)atoll(PQgetvalue(pgresult, j, 1));
-			key.reloid = reloid;
+			reloid    = atooid(PQgetvalue(pgresult, j, 0));
+			tableSize = (Size)atoll(PQgetvalue(pgresult, j, 1));
+			entry     = (ActiveTableEntryCombined *)hash_search(local_table_stats_map, &reloid, HASH_ENTER, &found);
+
 			/* for diskquota extension version is 1.0, pgresult doesn't contain segid */
 			if (PQnfields(pgresult) == 3)
 			{
 				/* get the segid, tablesize for each table */
-				segId     = atoi(PQgetvalue(pgresult, j, 2));
-				key.segid = segId;
-				entry     = (DiskQuotaActiveTableEntry *)hash_search(local_table_stats_map, &key, HASH_ENTER, &found);
-
-				if (!found)
-				{
-					/* receive table size info from the first segment */
-					entry->reloid = reloid;
-					entry->segid  = segId;
-				}
-				entry->tablesize = tableSize;
+				segId                       = atoi(PQgetvalue(pgresult, j, 2));
+				entry->tablesize[segId + 1] = tableSize;
 			}
 
 			/* when segid is -1, the tablesize is the sum of tablesize of master and all segments */
-			key.segid = -1;
```

Comment:
Since this line is removed, update please the comment above. Write something about entry->tablesize[0] instead of segid = -1

Reply:
updated